mllib/src/test/scala/org/apache/spark/ml/classification/NaiveBayesSuite.scala

```diff
@@ -150,6 +150,75 @@ class NaiveBayesSuite extends SparkFunSuite with MLlibTestSparkContext with Defa
     validateProbabilities(featureAndProbabilities, model, "multinomial")
   }
 
+  test("Naive Bayes Multinomial with weighted samples") {
+    val (dataset, weightedDataset) = {
+      val nPoints = 1000
+      val piArray = Array(0.5, 0.1, 0.4).map(math.log)
+      val thetaArray = Array(
+        Array(0.70, 0.10, 0.10, 0.10), // label 0
+        Array(0.10, 0.70, 0.10, 0.10), // label 1
+        Array(0.10, 0.10, 0.70, 0.10) // label 2
+      ).map(_.map(math.log))
+      val pi = Vectors.dense(piArray)
+      val theta = new DenseMatrix(3, 4, thetaArray.flatten, true)
+
+      val testData = generateNaiveBayesInput(piArray, thetaArray, nPoints, 42, "multinomial")
+
+      // Let's over-sample the label-1 samples twice, label-2 samples triple.
+      val data1 = testData.flatMap { case labeledPoint: LabeledPoint =>
+        labeledPoint.label match {
+          case 0.0 => Iterator(labeledPoint)
+          case 1.0 => Iterator(labeledPoint, labeledPoint)
+          case 2.0 => Iterator(labeledPoint, labeledPoint, labeledPoint)
+        }
+      }
+
+      val rnd = new Random(8392)
+      val data2 = testData.flatMap { case LabeledPoint(label: Double, features: Vector) =>
```

Comment:
I submitted a pr to your pr, with the weighted tests. (Hopefully I've done that correctly). Actually, I also think it is nice to test a case where the majority of the samples are outliers, but have small weights so they should not affect the predictions. This is semi-automated in MLUtils, but since NaiveBayes requires a certain type of features (0/1 in some cases) I don't think it integrates nicely yet. I think we should create a JIRA to automate weighted testing where we can think about this all together. For now, this test should be sufficient.